From dc776c2c7054b2ed7b26cdeb3dc4efdf842a6838 Mon Sep 17 00:00:00 2001
From: hyphop <art@khadas.com>
Date: Tue, 10 Mar 2020 23:21:40 +0900
Subject: [PATCH] env_save_default_to_fat

if uboot env file not found we can try to save default env for next usage
there we just define env_need_save env var like a flag 
for post save on last init stage before run_main_loop

---
 env/fat.c                          | 24 +++++++++++++++---------
 1 files changed

 
diff --git a/env/fat.c b/env/fat.c
index 9075147f..6ebcc8ec 100644
--- a/env/fat.c
+++ b/env/fat.c
@@ -110,6 +110,21 @@ static int env_fat_load(void)
 	}
 
 	err = file_fat_read(CONFIG_ENV_FAT_FILE, buf, CONFIG_ENV_SIZE);
+
+#define CONFIG_ENV_FAT_CREATE 1
+
+#ifdef CONFIG_ENV_FAT_CREATE
+//	env_set("env_source", env_get("boot_source"));
+//	printf("env_fat_load error: %d\n", err);
+//	file not found
+	if (err == -2) {
+		printf("\"%s\" not found on %s-%d:%d... ",
+			CONFIG_ENV_FAT_FILE, CONFIG_ENV_FAT_INTERFACE, dev, part );
+	    	env_set("env_need_save", "1");
+		return 0;
+	}
+#endif
+
 	if (err == -1) {
 		/*
 		 * This printf is embedded in the messages from env_save that
 	return env_import(buf, 1);
-- 
2.17.1

